protlib.py

```diff
@@ -173,7 +173,7 @@ def __init__(self, **settings):
             elif isinstance(self.length, int) and self.length < 0:
                 raise CError("length integer value must be positive")
         if "full_string" in settings and not isinstance(self, CString):
-            raise warn("full_string parameter has no meaning for {0} objects".format(self.__class__.__name__), CWarning)
+            warn("full_string parameter has no meaning for {0} objects".format(self.__class__.__name__), CWarning)
```

Comment:
This is a legit bugfix, although it should have never triggered for anyone, unless they specifically disabled ``CWarning`` as an exception raiser.  This is unrelated to Python 3.10 support, but I noticed it while looking through the ``raise`` statements in this codebase.